composable_kernel/include/tensor_operation/gridwise_2d_reduction_blockwise.hpp

```diff
@@ -0,0 +1,863 @@
+
+#include "threadwise_tensor_slice_transfer.hpp"
+
+namespace ck {
+
+template <typename GridwiseReduction,
+          bool NeedIndices,
+          typename InDataType,
+          typename OutDataType,
+          typename AccDataType,
+          typename In2dDescType,
+          typename Out1dDescType,
+          typename InElementwiseOperation,
+          typename OutElementwiseOperation>
+__global__ void kernel_reduce_blockwise(const In2dDescType in2dDesc,
+                                        const Out1dDescType out1dDesc,
+                                        const InElementwiseOperation in_elementwise_op,
+                                        const OutElementwiseOperation acc_elementwise_op,
+                                        AccDataType alpha,
+                                        const InDataType* const __restrict__ p_src_global,
+                                        OutDataType beta,
+                                        OutDataType* const __restrict__ p_dst_global,
+                                        const int* const __restrict__ ws_indices_global,
+                                        int* const __restrict__ indices_global)
+{
+    if constexpr(!NeedIndices)
+    {
+        GridwiseReduction::Run(in2dDesc,
+                               out1dDesc,
+                               in_elementwise_op,
+                               acc_elementwise_op,
+                               alpha,
+                               p_src_global,
+                               beta,
+                               p_dst_global,
+                               ws_indices_global,
+                               indices_global);
+    }
+    else
+    {
+        GridwiseReduction::RunWithIndices(in2dDesc,
+                                          out1dDesc,
+                                          in_elementwise_op,
+                                          acc_elementwise_op,
+                                          alpha,
+                                          p_src_global,
+                                          beta,
+                                          p_dst_global,
+                                          ws_indices_global,
+                                          indices_global);
+    };
+};
+
+template <typename GridwiseReduction,
+          bool NeedIndices,
+          typename InDataType,
+          typename OutDataType,
+          typename AccDataType,
+          typename In2dDescType,
+          typename Out1dDescType,
+          typename InElementwiseOperation,
+          typename OutElementwiseOperation>
+__global__ void
+kernel_reduce_blockwise_second_call(const In2dDescType in2dDesc,
+                                    const Out1dDescType out1dDesc,
+                                    const InElementwiseOperation in_elementwise_op,
+                                    const OutElementwiseOperation acc_elementwise_op,
+                                    AccDataType alpha,
```

Comment:
You can define a lambda function, instead of passing in alph and beta

Reply:
I have thought about that,  but since when `beta` is not zero,  loading data from device memory is needed and we could not put loading data in the lamba, and at least `beta` need be passed

Reply:
Please check this example of using functor for passing alpha/beta to GPU

https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/6d4450ef155c39af9ede2cd171be40ee06db9939/example/8_gemm_xdl_alpha_beta/gemm_xdl_alpha_beta.cpp#L33

https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/6d4450ef155c39af9ede2cd171be40ee06db9939/example/8_gemm_xdl_alpha_beta/gemm_xdl_alpha_beta.cpp#L208